src/Razor/test/Microsoft.NET.Sdk.Razor.Test/IntegrationTests/StaticAssetsIntegrationTest.cs

```diff
@@ -0,0 +1,296 @@
+            var assembly = Assert.ContainsEmbeddedResource(path, "Microsoft.AspNetCore.StaticAssets.xml");
+            using (var reader = new StreamReader(assembly))
+            {
+                var data = reader.ReadToEnd();
+                Assert.Equal(GetExpectedManifest(), data);
+            }
+        }
+
+        [Fact]
+        [InitializeTestProject("AppWithPackageAndP2PReference")]
+        public async Task GenerateAspNetCoreStaticAssetsManifest_IncrementalBuild_ReusesManifest()
+        {
+            // Arrange
+            // For some reason when using a custom package cache the imports won't get added on
+            // the initial restore, so we restore the packages ourselves.
+            await DotnetMSBuild("Restore");
+
+            var result = await DotnetMSBuild("GenerateAspNetCoreStaticAssetsManifest");
+
+            Assert.BuildPassed(result);
+
+            // GenerateAspNetCoreStaticAssetsManifest should generate the manifest and the cache.
+            Assert.FileExists(result, IntermediateOutputPath, "Microsoft.AspNetCore.StaticAssets.xml");
+            Assert.FileExists(result, IntermediateOutputPath, "Microsoft.AspNetCore.StaticAssets.cache");
+
+            var directoryPath = Path.Combine(result.Project.DirectoryPath, IntermediateOutputPath);
+            var thumbPrints = new Dictionary<string, FileThumbPrint>();
+            var thumbPrintFiles = new[]
+            {
+                Path.Combine(directoryPath, "Microsoft.AspNetCore.StaticAssets.xml"),
+                Path.Combine(directoryPath, "Microsoft.AspNetCore.StaticAssets.cache"),
+            };
+
+            foreach (var file in thumbPrintFiles)
+            {
+                var thumbprint = GetThumbPrint(file);
+                thumbPrints[file] = thumbprint;
+            }
+
+            // Act
+            var incremental = await DotnetMSBuild("GenerateAspNetCoreStaticAssetsManifest");
+
+            // Assert
+            Assert.BuildPassed(incremental);
+
+            foreach (var file in thumbPrintFiles)
+            {
+                var thumbprint = GetThumbPrint(file);
+                Assert.Equal(thumbPrints[file], thumbprint);
+            }
+        }
+
+        public Task InitializeAsync()
+        {
+            return PackageTestProjects.PackAsync(Output);
+        }
+
+        public Task DisposeAsync()
+        {
+            return Task.CompletedTask;
+        }
+    }
+
+    public class PackageTestProjectsFixture
+    {
+        private bool _packed;
+
+        internal async Task PackAsync(ITestOutputHelper output)
+        {
+            if (_packed)
+            {
+                return;
+            }
+
+            var projectsToPack = GetProjectsToPack();
+
+            foreach (var project in projectsToPack)
+            {
+                output.WriteLine(project);
+            }
+
+            foreach (var project in projectsToPack)
+            {
+                var psi = new ProcessStartInfo
```

Comment:
Can we re-use MSBuildProcess here - `dotnet msbuild /t:Pack`? It has a bunch of features around MSBuild process management that this does not capture.

Reply:
What exactly and why is it important?

Reply:
changed it, it would be good if you include explanations on why its important to do changes that seem trivial like this one. Specially around test setup